core/src/main/java/org/apache/carbondata/core/datastore/compression/GzipCompressor.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.datastore.compression;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.ShortBuffer;
+
+import org.apache.carbondata.core.util.ByteUtil;
+
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+
+public class GzipCompressor implements Compressor {
+
+  public GzipCompressor() {
+  }
+
+  @Override public String getName() {
+    return "gzip";
+  }
+
+  /*
+   * Method called for compressing the data and
+   * return a byte array
+   */
+  private byte[] compressData(byte[] data) {
+
+    ByteArrayOutputStream bt = new ByteArrayOutputStream();
+    try {
+      GzipCompressorOutputStream gzos = new GzipCompressorOutputStream(bt);
+      try {
+        gzos.write(data);
+      } catch (IOException e) {
+        e.printStackTrace();
+      } finally {
+        gzos.close();
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return bt.toByteArray();
```

Comment:
why `bt` is still open?

Reply:
ByteArrayOutputStream.close() does nothing. It's implementation in java is like this:

public void close() throws IOException {
}

I can close it but I'll have to copy the stream to byte Array and return that byte array which can be a costly operation.